crypto/dh/dh_pmeth.c

```diff
@@ -131,24 +141,19 @@ static int pkey_dh_copy(EVP_PKEY_CTX *dst, EVP_PKEY_CTX *src)
     dctx->kdf_type = sctx->kdf_type;
     dctx->kdf_oid = OBJ_dup(sctx->kdf_oid);
     if (!dctx->kdf_oid)
-        return 0;
+        goto err;
     dctx->kdf_md = sctx->kdf_md;
     if (dctx->kdf_ukm) {
         dctx->kdf_ukm = OPENSSL_memdup(sctx->kdf_ukm, sctx->kdf_ukmlen);
+        if (!dctx->kdf_ukm)
```

Comment:
while I am making patch I found a weird thing.

     if (dctx->kdf_ukm) {
          dctx->kdf_ukm = OPENSSL_memdup(sctx->kdf_ukm, sctx->kdf_ukmlen);

what does if(dctx->kdf_ukm) mean?
Actually it is dead code because dctx->kdf_ukm is from pkey_dh_init, which is initialized by 0.
Did I wrongly understand something?